iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/mempool_introspection.inl

```diff
@@ -196,7 +196,7 @@ void MemPoolIntrospection<MemoryManager, SegmentManager, PublisherPort>::copyMem
     for (uint32_t i = 0; i < numOfMemPools; ++i)
     {
         auto src = memoryManager.getMemPoolInfo(i);
-        auto& dst = dest[static_cast<int>(i)];
+        auto& dst = dest[static_cast<uint32_t>(i)];
```

Comment:
please use register size when indexing an array. In our case that would be 64 bit.